### PR TITLE
Added speed and cadence sensors

### PR DESCRIPTION
**Pull request by antoniodgonzalez**

### CODE REVIEW

File: src/sensors/cadence-sensors.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    BatteryVoltage?: number;
+    BatteryStatus?: 'New' | 'Good' | 'Ok' | 'Low' | 'Critical' | 'Invalid';
+    Motion?: boolean;
+    Rssi: number;
+    Threshold: number;
+}
+
+const DEVICE_TYPE = 0x7a;
+const PROFILE = 'CAD';
+const PERIOD = 8182;
```
Comment (gdoumen) [inline src/sensors/cadence-sensors.ts:36]: According to the spec, PERIOD should be 8102

The weird thing is that in the original implementation (wich was forked into gdoumen/ant-plus) the value i set to 8086

File: src/sensors/speed-sensors.ts [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    BatteryVoltage?: number;
+    BatteryStatus?: 'New' | 'Good' | 'Ok' | 'Low' | 'Critical' | 'Invalid';
+    Motion?: boolean;
+    Rssi: number;
+    Threshold: number;
+}
+
+const DEVICE_TYPE = 0x7b;
+const PROFILE = 'SPD';
+const PERIOD = 8182;
```
Comment (gdoumen) [inline src/sensors/speed-sensors.ts:37]: According to the spec, PERIOD should be 8118

The weird thing is that in the original implementation (wich was forked into gdoumen/ant-plus) the value is set to 8086

File: src/sensors/speed-sensors.ts [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+            period: PERIOD,
+            frequency: 57,
+        };
+    }
+
+    onEvent(data: Buffer) {
+        return;
+    }
+
+    wheelCircumference: number = 2.118; //This is my 700c wheel, just using as default
```
Comment (gdoumen) [inline src/sensors/speed-sensors.ts:64]: maybe we should move the member declaration to the top of the class and introduce a constant DEFAULT_WHEEL_CIRCUMFERECE=2.118  next to the other constants

File: src/sensors/speed-cadence-sensors.ts [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    CumulativeSpeedRevolutionCount: number;
+    CalculatedCadence: number;
+    CalculatedDistance: number;
+    CalculatedSpeed: number;
+    Rssi: number;
+    Threshold: number;
+}
+
+const DEVICE_TYPE = 0x79;
+const PROFILE = 'S&C';
```
Comment (gdoumen) [inline src/sensors/speed-cadence-sensors.ts:28]: I know that I told you to stick to the names in the ANT+ specs. However, in this particular case, I was asking myself if we could not end up in encoding issues with the & sign.

So would could consider to use SCS (**S**peed and **C**adence **S**ensor)

File: src/sensors/speed-cadence-sensors.ts [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+const PERIOD = 8182;
```
Comment (gdoumen) [inline src/sensors/speed-cadence-sensors.ts:29]: According to the spec, PERIOD is 8086

**Review by gdoumen — CHANGES_REQUESTED** [review]
Some of the constants are not set correctly. 

This might lead to a 'drift', i.e. the sensor might receive the values at the beginning, but might run into timing issues later

File: src/sensors/speed-cadence-sensors.ts (hunk 4 above)
Comment (antoniodgonzalez) [reply src/sensors/speed-cadence-sensors.ts:28]: I'm not fully convinced with using the Sensor suffix for consistence reasong, as is not used by other names. But I cannot find a better name.

**Review by gdoumen — APPROVED** [review]

File: src/sensors/speed-cadence-sensors.ts [hunk 6]
```diff
@@ -26,7 +26,7 @@ export class SpeedCadenceSensorState {
 
 const DEVICE_TYPE = 0x79;
 const PROFILE = 'S&C';
```
Comment (gdoumen) [inline src/sensors/speed-cadence-sensors.ts]: you still have left "S&C" in here, but in the samples you were using "SC"

**Review by gdoumen — CHANGES_REQUESTED** [review]

**Review by gdoumen — APPROVED** [review]

**Pull request merged**